tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1128,18 +1128,8 @@ tensorflow::Status ConvertConst(Converter& ctx,
   } else if (!weights_tensor.tensor_content().empty()) {
     VLOG(2) << "TENSOR!!!" << node_def.name();
     const auto& content = weights_tensor.tensor_content();
-
-    std::vector<char> values;
```

Comment:
Doesn't this value go out of scope right after this function?
Later when we get the value into TRT, it's pointing at garbage memory.

Why would we want to copy the memory out? TF owns the weights until the very end of the conversion, while TRT reads from and never writes to it during conversion. It seems safe.

Reply:
You're right that it will go out of scope, but before that in TRT_ShapedWeights we made a copy of the value (line 1142). Actually I'm not sure how memory management works in TRT, so I made a copy every time when a new TRT_ShapedWeights is copied from an old one (see the copy constructor). Is it enough?

And you're right that theoretically we don't need the copy. The main problem is, internally weights_tensor.tensor_content() is not a string, it's a class of complete different type having complete different APIs compared to tensorflow::string, so calling content.data() is invalid and won't compile. The only way to work around this is to call port::CopyToArray(), which internally accepts a different type of input (the internal type of 'content').

It's not ideal but that's the way how TF works. :( So please revert the changes here.

Reply:
Missed the part earlier, thought that owned_value was a pointer. TRT requires only that the data pointer is valid when we construct the network. So keeping a copy should work.
It must be one of the copy we are making that went wrong. The old code is not giving us wrong output from the converted network. Back onto investigation...

Reply:
There's some memory issue with the owned_value_. Data gets corrupted among many copies there. Moreover, we shouldn't need to make more than one copy of each weight anyway.
We allocate temporary memory per weight during the whole scope of the conversion. And pass pointers around.

Removed the std::vector for that part and port::CopyToArray throw data into our temporary memory. Please let me know if it makes internal build happy :)

Next PR will remove owned_value and clean it up.

Reply:
Thanks for the fix! I didn't notice that there was an get_temp_weights() method. Please go ahead to clean it up, as long as we're using CopyToArray it should be fine.

Reply:
Done :)